### PR TITLE
Uses OpenAI API

### PR DESCRIPTION
**Pull request by tim-macphail**
## Overview:
* New route `/premium-files` acts the same as `/files` but uses the openai api. The client uses the new route, but the old one still exists.

* #235 will have to adapt to this, since the api generated responses are much more valuable and have a greater reason to be cached.

* Includes refactoring to `file_handler.py` to make things more readable

* I've already put the necessary environment variables into the lambda function via the aws console for this to work once it's merged (hopefully).

* OpenAI billing profile is on free trial, which expires in `min(1 month, $18)`. Long term solution TBD.

## Future improvements:
- Split `full_text` on a more logical boundary than simply every 4096 tokens
- ~Send API calls with chunks concurrently to boost performance~
- Cache responses to improve performance and reduce cost #235 
- Potential fine tuning of prompt/hyperparams

### CODE REVIEW

File: client/src/pages/Review/Review.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ const Review = () => {
       formData.append("outline_file", file as File);
 
       await axios
-        .post("/files", formData, {
+        .post("/premium-files", formData, {
```
Comment (harsweet) [inline client/src/pages/Review/Review.tsx:89]: So we'll use the endpoint powered by ChatGPT as the only one till we have the front-end to support both?

File: server/src/handlers/system-prompt.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+You are an algorithm designed to extract deadlines from text.
```
Comment (harsweet) [inline server/src/handlers/system-prompt.txt:1]: Nice:)

**Review by harsweet — APPROVED** [review]
Great Work!!

File: server/src/handlers/file_handler.py [hunk 3]
```diff
@@ -95,45 +134,43 @@ def get_course_key(pdf: pdfplumber.pdf.PDF):
...
 
+    if course_code and course_number:
+        course_key = f"{course_code} {course_number}"
+        course_info = COURSE_INFOS[course_key]
+        if course_info is not None:
+            course = {**course, **course_info}
+    return course
+
+
+def get_assessments_from_tables(pdf: pdfplumber.pdf.PDF):
```
[the remark this answers is not in the document]
Comment (tim-macphail) [reply server/src/handlers/file_handler.py:153]: Yes... I was thinking the same thing but I was unsure the best choice on what to name files / where to put them. Do you have any thoughts on this architecture?

File: server/.python-version
Comment (tim-macphail) [inline server/.python-version:1]: For `pyenv` python version manager

**Comment by cloudyyoung** [comment]
@tim-macphail Seem to not include newly introduced libraries in `requirements.txt` in this PR

**Comment by tim-macphail** [comment]
> @tim-macphail Seem to not include newly introduced libraries in `requirements.txt` in this PR. Can you also include a `.env.example` file?

The diff view in gh for requirements.txt is confusing me. I guess the tests still passed because theres no tests for the new route. But also it's working on production just fine? 

What is `.env.example`? I sent everyone the .env file on discord

**Comment by cloudyyoung** [comment]
> > @tim-macphail Seem to not include newly introduced libraries in `requirements.txt` in this PR. Can you also include a `.env.example` file?
> 
> The diff view in gh for requirements.txt is confusing me. I guess the tests still passed because theres no tests for the new route. But also it's working on production just fine?
> 
> What is `.env.example`? I sent everyone the .env file on discord

I know, but still. https://www.codementor.io/@parthibakumarmurugesan/what-is-env-how-to-set-up-and-run-a-env-file-in-node-1pnyxw9yxj

**Pull request merged**